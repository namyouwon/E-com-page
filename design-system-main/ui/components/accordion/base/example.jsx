// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

import React from 'react';
import uniqueId from 'lodash.uniqueid';
import Accordion, { AccordionSection } from '../';
import { ActionOverflow } from '../../../components/menus/action-overflow/example';
import Card from '../../cards/';

export const AccordionExample = () => (
  <Accordion>
    <AccordionSection
      summary="Accordion summary"
      isOpen
      referenceId={uniqueId('referenceId-')}
    >
      Accordion details - A
    </AccordionSection>
    <AccordionSection
      summary="Accordion summary"
      referenceId={uniqueId('referenceId-')}
    >
      Accordion details - B
    </AccordionSection>
    <AccordionSection
      summary="Accordion summary"
      referenceId={uniqueId('referenceId-')}
    >
      Accordion details - C
    </AccordionSection>
  </Accordion>
);

export default <AccordionExample />;

export let states = [
  {
    id: 'section-one-open',
    label: 'Section one open',
    element: (
      <Accordion>
        <AccordionSection
          summary="Accordion summary"
          isOpen
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - A
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - B
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - C
        </AccordionSection>
      </Accordion>
    )
  },
  {
    id: 'section-two-open',
    label: 'Section two open',
    element: (
      <Accordion>
        <AccordionSection
          summary="Accordion summary"
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - A
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          isOpen
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - B
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - C
        </AccordionSection>
      </Accordion>
    )
  },
  {
    id: 'section-three-open',
    label: 'Section three open',
    element: (
      <Accordion>
        <AccordionSection
          summary="Accordion summary"
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - A
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - B
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          isOpen
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - C
        </AccordionSection>
      </Accordion>
    )
  },
  {
    id: 'multiple-sections-open',
    label: 'Multiple sections open',
    element: (
      <Accordion>
        <AccordionSection
          summary="Accordion summary"
          isOpen
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - A
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - B
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          isOpen
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - C
        </AccordionSection>
      </Accordion>
    )
  }
];

export let examples = [
  {
    id: 'long-heading-truncated',
    label: 'Long Section Heading with Truncation for larger screens',
    demoStyles: 'max-width: 500px;',
    storybookStyles: true,
    element: (
      <Card>
        <Accordion>
          <AccordionSection
            summary="Accordion summary Accordion summary Accordion summary Accordion summary Accordion summary Accordion summary Accordion summary Accordion summary Accordion summary Accordion summary"
            isOpen
            referenceId={uniqueId('referenceId-')}
          >
            Accordion details - A
          </AccordionSection>
          <AccordionSection
            summary="Accordion summary"
            referenceId={uniqueId('referenceId-')}
          >
            Accordion details - B
          </AccordionSection>
          <AccordionSection
            summary="Accordion summary"
            referenceId={uniqueId('referenceId-')}
          >
            Accordion details - C
          </AccordionSection>
        </Accordion>
      </Card>
    )
  },
  {
    id: 'nested-accordions',
    label: 'Nested Accordions',
    element: (
      <Accordion>
        <AccordionSection
          summary="Accordion summary"
          isOpen
          referenceId={uniqueId('referenceId-')}
        >
          <Accordion>
            <AccordionSection
              summary="Nested accordion summary"
              isOpen
              referenceId={uniqueId('referenceId-')}
            >
              Accordion details - A
            </AccordionSection>
            <AccordionSection
              summary="Nested accordion summary"
              referenceId={uniqueId('referenceId-')}
            >
              Accordion details - B
            </AccordionSection>
          </Accordion>
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - B
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - C
        </AccordionSection>
      </Accordion>
    )
  },
  {
    id: 'styled',
    label: 'Wrapped in Card',
    element: (
      <div className="slds-card">
        <Accordion>
          <AccordionSection
            summary="Accordion summary"
            isOpen
            referenceId={uniqueId('referenceId-')}
          >
            Accordion details - A
          </AccordionSection>
          <AccordionSection
            summary="Accordion summary"
            referenceId={uniqueId('referenceId-')}
          >
            Accordion details - B
          </AccordionSection>
          <AccordionSection
            summary="Accordion summary"
            referenceId={uniqueId('referenceId-')}
          >
            Accordion details - C
          </AccordionSection>
        </Accordion>
      </div>
    )
  },
  {
    id: 'has-action-overflow',
    label: 'Has Action Overflow Menu',
    element: (
      <Accordion>
        <AccordionSection
          summary="Accordion summary"
          actionMenu={
            <ActionOverflow position="right" dropdownIsOpen={false} />
          }
          isOpen
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - A
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          actionMenu={
            <ActionOverflow position="right" dropdownIsOpen={false} />
          }
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - B
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          actionMenu={
            <ActionOverflow position="right" dropdownIsOpen={false} />
          }
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - C
        </AccordionSection>
      </Accordion>
    )
  },
  {
    id: 'section-background-color',
    label: 'Accordion Section - Background Color',
    demoStyles: '--slds-c-accordion-summary-color-background: lightblue',
    storybookStyles: true,
    element: (
      <Accordion>
        <AccordionSection
          summary="Accordion summary"
          isOpen
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - A
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - B
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - C
        </AccordionSection>
      </Accordion>
    )
  },
  {
    id: 'section-background-color',
    label: 'Accordion Section - Text Color',
    demoStyles:
      '--slds-c-accordion-heading-text-color: green; --slds-c-accordion-heading-text-color-hover: red',
    storybookStyles: true,
    element: (
      <Accordion>
        <AccordionSection
          summary="Accordion summary"
          isOpen
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - A
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - B
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - C
        </AccordionSection>
      </Accordion>
    )
  }
];
