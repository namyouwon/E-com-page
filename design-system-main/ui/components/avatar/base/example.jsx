// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

import React from 'react';
import { Avatar, AvatarIcon, AvatarInitials } from '../';

/// ///////////////////////////////////////////
// Export
/// ///////////////////////////////////////////

export { Avatar };

export default [
  {
    id: 'default',
    label: 'Default',
    element: (
      <Avatar>
        <img
          alt="Person name"
          src="/assets/images/avatar2.jpg"
          title="Person name"
        />
      </Avatar>
    )
  }
];

export let examples = [
  {
    id: 'circle',
    label: 'Circle',
    element: (
      <Avatar className="slds-avatar_circle">
        <img
          alt="Person name"
          src="/assets/images/avatar2.jpg"
          title="Person name"
        />
      </Avatar>
    )
  },
  {
    id: 'profile-icon',
    label: 'Profile Avatar Image',
    element: (
      <Avatar className="slds-avatar_profile-image-large">
        <span className="slds-assistive-text">Person name</span>
      </Avatar>
    )
  },
  {
    id: 'group-icon',
    label: 'Group Avatar Image',
    element: (
      <Avatar className="slds-avatar_group-image-large">
        <span className="slds-assistive-text">Group name</span>
      </Avatar>
    )
  },
  {
    id: 'fallback-user-initials',
    label: 'Fallback - User initials',
    element: (
      <Avatar className="slds-avatar_circle">
        <AvatarInitials
          iconClassName="slds-icon-standard-user"
          title="person name"
          initials="AB"
        />
      </Avatar>
    )
  },
  {
    id: 'fallback-user-icon',
    label: 'Fallback - User icon',
    element: (
      <Avatar className="slds-avatar_circle">
        <AvatarIcon symbol="user" />
      </Avatar>
    )
  },
  {
    id: 'fallback-entity-initials',
    label: 'Fallback - Entity Initials',
    element: (
      <Avatar>
        <AvatarInitials
          iconClassName="slds-icon-standard-account"
          title="Company name"
          initials="Ac"
        />
      </Avatar>
    )
  },
  {
    id: 'fallback-entity-icon',
    label: 'Fallback - Entity Icon',
    element: (
      <Avatar>
        <AvatarIcon />
      </Avatar>
    )
  },
  {
    id: 'fallback-user-initials-inverse',
    label: 'Fallback - User Initials Inversed',
    element: (
      <Avatar className="slds-avatar_circle">
        <AvatarInitials title="Person name" initials="AB" isInverse />
      </Avatar>
    )
  },
  {
    id: 'x-small-image',
    label: 'X-Small Image',
    element: (
      <Avatar className="slds-avatar_circle slds-avatar_x-small">
        <img
          alt="Person name"
          src="/assets/images/avatar2.jpg"
          title="Person name"
        />
      </Avatar>
    )
  },
  {
    id: 'small-image',
    label: 'Small Image',
    element: (
      <Avatar className="slds-avatar_circle slds-avatar_small">
        <img
          alt="Person name"
          src="/assets/images/avatar2.jpg"
          title="Person name"
        />
      </Avatar>
    )
  },
  {
    id: 'medium-image',
    label: 'Medium Image',
    element: (
      <Avatar className="slds-avatar_circle slds-avatar_medium">
        <img
          alt="Person name"
          src="/assets/images/avatar2.jpg"
          title="Person name"
        />
      </Avatar>
    )
  },
  {
    id: 'large-image',
    label: 'Large Image',
    element: (
      <Avatar className="slds-avatar_circle slds-avatar_large">
        <img
          alt="Person name"
          src="/assets/images/avatar2.jpg"
          title="Person name"
        />
      </Avatar>
    )
  },

  {
    id: 'linked-avatar',
    label: 'Wrapped in a link',
    element: (
      <a href="#" onClick={e => e.preventDefault()}>
        <Avatar>
          <img
            alt="Person name"
            src="/assets/images/avatar2.jpg"
            title="User avatar"
          />
        </Avatar>
      </a>
    )
  }
];
