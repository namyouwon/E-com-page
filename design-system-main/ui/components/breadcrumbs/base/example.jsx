// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

import React from 'react';
import BreadCrumbs from '../index.react';
import ButtonIcon from '../../button-icons/';
import {
  Trigger,
  Menu,
  MenuList,
  MenuItem
} from '../../menus/dropdown/example';
const { Crumb } = BreadCrumbs;

const moreIcon = (
  <ButtonIcon
    className="slds-button_icon-border-filled slds-button_icon-x-small"
    symbol="threedots"
    assistiveText="Show More"
    aria-haspopup="true"
    title="Show More"
  />
);

let BreadcrumbMenu = props => (
  <Trigger isOpen triggerIcon={moreIcon}>
    <Menu className="slds-dropdown_left slds-dropdown_actions">
      <MenuList>
        <MenuItem tabIndex="0">Menu Item One</MenuItem>
        <MenuItem>Menu Item Two</MenuItem>
        <MenuItem>Menu Item Three</MenuItem>
      </MenuList>
    </Menu>
  </Trigger>
);

export const Context = props => (
  <div className="demo-only" style={{ height: '150px' }}>
    {props.children}
  </div>
);

export const BreadcrumbsBase = ({ kineticsEnabled }) => (
  <BreadCrumbs
    kxScopeBreadcrumbsItem={kineticsEnabled}
    kxTypeUnderline={kineticsEnabled}
  >
    <Crumb>Parent Entity</Crumb>
    <Crumb>Parent Record Name</Crumb>
  </BreadCrumbs>
);

/* for snapshot tests */
export const BreadcrumbsWithOverflow = ({ kineticsEnabled }) => (
  <BreadCrumbs
    kxScopeBreadcrumbsItem={kineticsEnabled}
    kxTypeUnderline={kineticsEnabled}
    listClassNames="slds-grid_vertical-align-center"
  >
    <Crumb hasMenu>
      <BreadcrumbMenu />
    </Crumb>
    <Crumb>Parent Entity</Crumb>
    <Crumb>Parent Record Name</Crumb>
  </BreadCrumbs>
);

/// ///////////////////////////////////////////
// Export
/// ///////////////////////////////////////////

/* export for snapshot tests */
export default <BreadcrumbsBase />;

/* export for site examples */
export let states = [
  {
    id: 'breadcrumbs-overflow',
    label: 'With Overflow Menu',
    demoStyles: 'height: 200px;',
    storybookStyles: true,
    element: <BreadcrumbsWithOverflow />
  }
];

export let examples = [
  {
    id: 'kinetics-breadcrumbs-base',
    label: 'Base (Kinetics)',
    element: <BreadcrumbsBase kineticsEnabled />
  },
  {
    id: 'kinetics-breadcrumbs-overflow',
    label: 'With Overflow Menu (Kinetics)',
    demoStyles: 'height: 200px;',
    storybookStyles: true,
    element: <BreadcrumbsWithOverflow kineticsEnabled />
  }
];
