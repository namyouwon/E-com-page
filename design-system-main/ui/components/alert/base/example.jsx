// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

import React from 'react';
import Alert from '../';
import { UtilityIcon } from '../../icons/base/example';
import EmptyLink from '../../../shared/empty-link/';
import Heading from '../../../shared/heading/';

export default [
  {
    id: 'default',
    label: 'Default',
    element: (
      <Alert>
        <UtilityIcon
          containerClassName="slds-m-right_x-small"
          className="slds-icon_x-small"
          assistiveText={false}
          symbol="user"
        />
        <Heading level="2">
          Logged in as John Smith (johnsmith@acme.com).{' '}
          <EmptyLink>Log out</EmptyLink>
        </Heading>
      </Alert>
    )
  }
];

export let states = [
  {
    id: 'theme-info',
    label: 'Default (Theme Class  - Deprecated)',
    element: (
      <Alert isLegacy>
        <UtilityIcon
          containerClassName="slds-m-right_x-small"
          className="slds-icon_x-small"
          assistiveText={false}
          symbol="user"
        />
        <Heading level="2">
          Logged in as John Smith (johnsmith@acme.com).{' '}
          <EmptyLink>Log out</EmptyLink>
        </Heading>
      </Alert>
    )
  },
  {
    id: 'warning',
    label: 'Warning',
    element: (
      <Alert type="warning">
        <UtilityIcon
          containerClassName="slds-m-right_x-small"
          className="slds-icon_x-small"
          assistiveText={false}
          symbol="warning"
        />
        <Heading level="2">
          Your browser is outdated. Your Salesforce experience may be degraded.{' '}
          <EmptyLink>More Information</EmptyLink>
        </Heading>
      </Alert>
    )
  },
  {
    id: 'theme-warning',
    label: 'Warning (Theme Class  - Deprecated)',
    element: (
      <Alert type="warning" isLegacy>
        <UtilityIcon
          containerClassName="slds-m-right_x-small"
          className="slds-icon_x-small"
          assistiveText={false}
          symbol="warning"
        />
        <Heading level="2">
          Your browser is outdated. Your Salesforce experience may be degraded.{' '}
          <EmptyLink>More Information</EmptyLink>
        </Heading>
      </Alert>
    )
  },
  {
    id: 'error',
    label: 'Error',
    element: (
      <Alert type="error">
        <UtilityIcon
          containerClassName="slds-m-right_x-small"
          className="slds-icon_x-small"
          assistiveText={false}
          symbol="error"
        />
        <Heading level="2">
          Your browser is currently not supported. Your Salesforce may be
          degraded. <EmptyLink>More Information</EmptyLink>
        </Heading>
      </Alert>
    )
  },
  {
    id: 'theme-error',
    label: 'Error (Theme Class  - Deprecated)',
    element: (
      <Alert type="error" isLegacy>
        <UtilityIcon
          containerClassName="slds-m-right_x-small"
          className="slds-icon_x-small"
          assistiveText={false}
          symbol="error"
        />
        <Heading level="2">
          Your browser is currently not supported. Your Salesforce may be
          degraded. <EmptyLink>More Information</EmptyLink>
        </Heading>
      </Alert>
    )
  },
  {
    id: 'offline',
    label: 'Offline',
    element: (
      <Alert type="offline">
        <UtilityIcon
          containerClassName="slds-m-right_x-small"
          className="slds-icon_x-small"
          assistiveText={false}
          symbol="offline"
        />
        <Heading level="2">
          You are in offline mode. <EmptyLink>More Information</EmptyLink>
        </Heading>
      </Alert>
    )
  },
  {
    id: 'theme-offline',
    label: 'Offline (Theme Class  - Deprecated)',
    element: (
      <Alert type="offline" isLegacy>
        <UtilityIcon
          containerClassName="slds-m-right_x-small"
          className="slds-icon_x-small"
          assistiveText={false}
          symbol="offline"
        />
        <Heading level="2">
          You are in offline mode. <EmptyLink>More Information</EmptyLink>
        </Heading>
      </Alert>
    )
  }
];
