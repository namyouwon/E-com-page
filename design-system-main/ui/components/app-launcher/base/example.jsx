// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

import React from 'react';
import SvgIcon from '../../../shared/svg-icon';
import {
  Modal,
  ModalHeader,
  ModalContent,
  Backdrop
} from '../../modals/base/example';
import {
  Section,
  SectionContent,
  SectionTitle,
  SectionTitleAction
} from '../../expandable-section/base/example';
import { Avatar } from '../../avatar/base/example';
import { ButtonIconStateful } from '../../button-icons/stateful/example';
import { Tooltip } from '../../tooltips/base/example';
import classNames from 'classnames';

const exampleDemoStyles = 'height: 800px;';

export const Context = props => (
  <div style={{ height: '800px' }}>{props.children}</div>
);

/// ///////////////////////////////////////////
// Partial(s)
/// ///////////////////////////////////////////
export let AppLauncherModal = props => (
  <Modal
    className="slds-modal_large slds-app-launcher"
    aria-labelledby="header43"
  >
    <ModalHeader className="slds-grid slds-grid_align-spread slds-grid_vertical-align-center">
      <h2 id="header43" className="slds-text-heading_medium">
        App launcher
      </h2>
      <div className="slds-app-launcher__header-search">
        <div className="slds-form-element">
          <label
            htmlFor="app-launcher-search"
            className="slds-form-element__label slds-assistive-text"
          >
            Find an app
          </label>
          <div className="slds-form-element__control slds-input-has-icon slds-input-has-icon_left">
            <SvgIcon
              className="slds-input__icon"
              sprite="utility"
              symbol="search"
            />
            <input
              type="search"
              className="slds-input"
              id="app-launcher-search"
              placeholder="Find an app"
              defaultValue={props.searchTerm}
            />
          </div>
        </div>
      </div>
      <button className="slds-button slds-button_neutral">App Exchange</button>
    </ModalHeader>
    <ModalContent className="slds-app-launcher__content slds-p-around_medium">
      <Section className="slds-is-open">
        <SectionTitle>
          <SectionTitleAction isOpen referenceId="appsContent">
            All Apps
          </SectionTitleAction>
        </SectionTitle>
        <SectionContent isOpen referenceId="appsContent">
          <div
            className="slds-assistive-text"
            id="drag-live-region"
            aria-live="assertive"
          >
            {props.dragDropLiveRegion}
          </div>
          <ul className="slds-grid slds-grid_pull-padded slds-wrap">
            {props.appTiles.map((tile, i) => {
              return (
                <li
                  className="slds-p-horizontal_small slds-size_1-of-1 slds-medium-size_1-of-3"
                  key={i}
                >
                  <AppLauncherTile
                    figureClass={tile.figureClass}
                    grabbed={tile.grabbed}
                    objectInitials={tile.initials}
                    label={tile.label}
                    description={tile.description}
                    hasTooltip={tile.hasTooltip}
                    tooltipText={tile.tooltipText}
                    index={i}
                  />
                </li>
              );
            })}
          </ul>
        </SectionContent>
      </Section>
      <hr />
      <Section className="slds-is-open">
        <SectionTitle>
          <SectionTitleAction isOpen referenceId="itemsContent">
            All Items
          </SectionTitleAction>
        </SectionTitle>
        <SectionContent isOpen referenceId="itemsContent">
          <ul className="slds-grid slds-wrap">
            {props.itemTiles.map((tile, i) => {
              return (
                <li
                  className="slds-col_padded slds-p-vertical_xx-small slds-size_1-of-5"
                  key={i}
                >
                  <a
                    href="#"
                    className="slds-truncate"
                    title={tile.label}
                    onClick={e => e.preventDefault()}
                  >
                    {tile.label}
                  </a>
                </li>
              );
            })}
          </ul>
        </SectionContent>
      </Section>
    </ModalContent>
  </Modal>
);

export const AppLauncherTile = props => (
  <div
    draggable
    className={classNames(
      'slds-app-launcher__tile slds-text-link_reset slds-is-draggable',
      props.className,
      {
        'slds-is-grabbed': props.grabbed
      }
    )}
  >
    <div className="slds-app-launcher__tile-figure">
      <Avatar className="slds-avatar_large">
        <abbr
          className={classNames('slds-avatar__initials', props.figureClass)}
          title={props.label}
        >
          {props.objectInitials}
        </abbr>
      </Avatar>
      <div className="slds-m-top_xxx-small">
        <ButtonIconStateful
          aria-pressed={props.grabbed ? 'true' : 'false'}
          assistiveText="Reorder"
          selected={props.grabbed}
          symbol="rows"
          title="Reorder"
        />
      </div>
    </div>
    <div className="slds-app-launcher__tile-body">
      <a href="#" onClick={e => e.preventDefault()}>
        {props.label}
      </a>
      <p>
        {props.description}{' '}
        {props.hasTooltip && (
          <button
            aria-describedby={`help-${props.index}`}
            className="slds-button slds-button_reset slds-text-link"
          >
            More
          </button>
        )}
      </p>
      <Tooltip
        className={classNames('slds-nubbin_top-right', {
          'slds-hide': !props.tooltipText
        })}
        id={`help-${props.index}`}
        style={{ position: 'absolute', top: '80px', right: '30px' }}
      >
        {props.tooltipText}
      </Tooltip>
    </div>
  </div>
);

/// ///////////////////////////////////////////
// Export
/// ///////////////////////////////////////////

/*
 * DragDropId relates app launcher tile anchor aria-describedby to the div that holds the instructions for drag & drop
 */
export const itemTiles = [
  { label: 'Accounts', symbol: 'account' },
  { label: 'Announcements', symbol: 'announcement' },
  { label: 'Approvals', symbol: 'approval' },
  { label: 'Campaigns', symbol: 'campaign' },
  { label: 'Cases', symbol: 'case' },
  { label: 'Coaching', symbol: 'coaching' },
  { label: 'Contacts', symbol: 'contact' }
];

export const searchItemTiles = [
  {
    label: (
      <span>
        <mark>Sales</mark> Invoices
      </span>
    ),
    symbol: 'account'
  },
  {
    label: (
      <span>
        <mark>Sales</mark> Objects
      </span>
    ),
    symbol: 'announcement'
  }
];

/*
 * Tile data for All Apps section inside example.
 * Using data instead of hard-coding becase we want to move the position of one App Tile based on state (ie. moved, dropped).
 */
export const appTiles = [
  {
    description:
      'The primary internal Salesforce org. Used to run our online sales business...',
    hasTooltip: true,
    figureClass: 'slds-icon-custom-27',
    grabbed: false,
    initials: 'SC',
    label: 'Sales Cloud'
  },
  {
    description: 'Salesforce Marketing Cloud lets businesses of any size...',
    hasTooltip: true,
    figureClass: 'slds-icon-custom-59',
    grabbed: false,
    initials: 'MC',
    label: 'Marketing Cloud'
  },
  {
    description: 'Community for managing employee benefits and time off.',
    hasTooltip: false,
    figureClass: 'slds-icon-custom-10',
    grabbed: false,
    initials: 'HR',
    label: 'HR Concierge'
  },
  {
    description: 'Manage your finances across multiple financial platforms...',
    hasTooltip: true,
    figureClass: 'slds-icon-custom-6',
    grabbed: false,
    initials: 'MM',
    label: 'My Money'
  },
  {
    description:
      'The key to call center and contact center management is more...',
    hasTooltip: true,
    figureClass: 'slds-icon-custom-91',
    grabbed: false,
    initials: 'CC',
    label: 'Call Center'
  },
  {
    description:
      'Areas of Focus are used to track customer support for your...',
    hasTooltip: true,
    figureClass: 'slds-icon-custom-50',
    grabbed: false,
    initials: 'CS',
    label: 'Customer Support Community'
  }
];

export const appTilesSearch = [
  {
    description: 'The primary internal Salesforce org. Used to run our...',
    hasTooltip: true,
    figureClass: 'slds-icon-custom-27',
    grabbed: false,
    initials: 'SC',
    label: (
      <span>
        <mark>Sales</mark> Cloud
      </span>
    )
  }
];

/*
 * Helpers to position the App Tile for each state (ie. moved, dropped, etc.)
 */
const moveItemPosition = function(arr, from, to) {
  let copyArr = arr.slice(0);
  copyArr.splice(to, 0, copyArr.splice(from, 1)[0]);
  return copyArr;
};

export const appTilesGrabbed = appTiles.slice(0);
appTilesGrabbed[0] = Object.assign({}, appTilesGrabbed[0], { grabbed: true });
export const appTilesMoved = moveItemPosition(appTiles, 0, 2);
appTilesMoved[2] = Object.assign({}, appTilesMoved[2], { grabbed: true });
export const appTilesDropped = moveItemPosition(appTiles, 0, 3);

const defaultComponent = (
  <Backdrop>
    <AppLauncherModal
      appTiles={appTiles}
      dragDropLiveRegion=""
      itemTiles={itemTiles}
    />
  </Backdrop>
);

export default defaultComponent;

export let states = [
  {
    id: 'default',
    label: 'Default',
    demoStyles: exampleDemoStyles,
    storybookStyles: true,
    element: defaultComponent
  },
  {
    id: 'grabbed',
    label: 'Tile grabbed',
    demoStyles: exampleDemoStyles,
    storybookStyles: true,
    element: (
      <Backdrop>
        <AppLauncherModal
          appTiles={appTilesGrabbed}
          dragDropLiveRegion="Sales Cloud: current position 1 of 6. Use the up and down arrows to move this app"
          grabbed
          itemTiles={itemTiles}
        />
      </Backdrop>
    )
  },
  {
    id: 'moved',
    label: 'Tile moved in list',
    demoStyles: exampleDemoStyles,
    storybookStyles: true,
    element: (
      <Backdrop>
        <AppLauncherModal
          appTiles={appTilesMoved}
          dragDropLiveRegion="Sales Cloud: new position 3 of 6."
          itemTiles={itemTiles}
        />
      </Backdrop>
    )
  },
  {
    id: 'dropped',
    label: 'Tile dropped',
    demoStyles: exampleDemoStyles,
    storybookStyles: true,
    element: (
      <Backdrop>
        <AppLauncherModal
          appTiles={appTilesDropped}
          dragDropLiveRegion="Sales Cloud: final position 4 of 6."
          itemTiles={itemTiles}
        />
      </Backdrop>
    )
  },
  {
    id: 'search',
    label: 'Search',
    demoStyles: exampleDemoStyles,
    storybookStyles: true,
    element: (
      <Backdrop>
        <AppLauncherModal
          appTiles={appTilesSearch}
          dragDropLiveRegion=""
          itemTiles={searchItemTiles}
          searchTerm="sales"
        />
      </Backdrop>
    )
  },
  {
    id: 'tooltip',
    label: 'Tooltip',
    demoStyles: 'width: 360px; height: 13rem; position: relative; padding: 0;',
    storybookStyles: true,
    element: (
      <AppLauncherTile
        figureClass="slds-icon-custom-27"
        objectInitials="SC"
        label="Sales Cloud"
        description="The primary internal Salesforce org. Used to run our online sales business..."
        hasTooltip
        tooltipText="The primary internal Salesforce org. Used to run our online sales business and manage accounts."
        index="x"
      />
    )
  }
];
