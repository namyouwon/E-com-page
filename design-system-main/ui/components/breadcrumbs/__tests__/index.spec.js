// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license
/* eslint-env jest */

import React from 'react';
import createHelpers from '../../../../jest.helpers';
import { BreadcrumbsBase, BreadcrumbsWithOverflow } from '../base/example';

const { matchesMarkup } = createHelpers(__dirname);

it('renders a base breadcrumb', () => matchesMarkup(<BreadcrumbsBase />));

it('renders a overflow menu as the first breadcrumb', () =>
  matchesMarkup(<BreadcrumbsWithOverflow />));
