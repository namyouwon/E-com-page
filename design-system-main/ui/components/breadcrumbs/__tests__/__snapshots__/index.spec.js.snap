// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a base breadcrumb 1`] = `
<nav
  aria-label="Breadcrumbs"
  role="navigation"
>
  <ol
    className="slds-breadcrumb slds-list_horizontal slds-wrap"
  >
    <li
      className="slds-breadcrumb__item"
    >
      <a
        href="#"
        onClick={[Function]}
      >
        Parent Entity
      </a>
    </li>
    <li
      className="slds-breadcrumb__item"
    >
      <a
        href="#"
        onClick={[Function]}
      >
        Parent Record Name
      </a>
    </li>
  </ol>
</nav>
`;

exports[`renders a overflow menu as the first breadcrumb 1`] = `
<nav
  aria-label="Breadcrumbs"
  role="navigation"
>
  <ol
    className="slds-breadcrumb slds-list_horizontal slds-wrap slds-grid_vertical-align-center"
  >
    <li
      className="slds-breadcrumb__item"
    >
      <div
        className="slds-dropdown-trigger slds-dropdown-trigger_click slds-is-open"
      >
        <button
          aria-haspopup="true"
          className="slds-button slds-button_icon slds-button_icon-border-filled slds-button_icon-x-small"
          onClick={[Function]}
          title="Show More"
        >
          <svg
            aria-hidden={true}
            className="slds-button__icon"
          >
            <use
              xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#threedots"
            />
          </svg>
          <span
            className="slds-assistive-text"
          >
            Show More
          </span>
        </button>
        <div
          className="slds-dropdown slds-dropdown_left slds-dropdown_actions"
        >
          <ul
            className="slds-dropdown__list"
            role="menu"
          >
            <li
              className="slds-dropdown__item"
              role="presentation"
            >
              <a
                href="#"
                onClick={[Function]}
                role="menuitem"
                tabIndex="0"
              >
                <span
                  className="slds-truncate"
                  title="Menu Item One"
                >
                  Menu Item One
                </span>
              </a>
            </li>
            <li
              className="slds-dropdown__item"
              role="presentation"
            >
              <a
                href="#"
                onClick={[Function]}
                role="menuitem"
                tabIndex="-1"
              >
                <span
                  className="slds-truncate"
                  title="Menu Item Two"
                >
                  Menu Item Two
                </span>
              </a>
            </li>
            <li
              className="slds-dropdown__item"
              role="presentation"
            >
              <a
                href="#"
                onClick={[Function]}
                role="menuitem"
                tabIndex="-1"
              >
                <span
                  className="slds-truncate"
                  title="Menu Item Three"
                >
                  Menu Item Three
                </span>
              </a>
            </li>
          </ul>
        </div>
      </div>
    </li>
    <li
      className="slds-breadcrumb__item"
    >
      <a
        href="#"
        onClick={[Function]}
      >
        Parent Entity
      </a>
    </li>
    <li
      className="slds-breadcrumb__item"
    >
      <a
        href="#"
        onClick={[Function]}
      >
        Parent Record Name
      </a>
    </li>
  </ol>
</nav>
`;
